lintro/cli_utils/commands/setup.py

```
@@ -0,0 +1,503 @@
+from itertools import chain
+from pathlib import Path
+
+import click
+from rich.console import Console
+
+from lintro.tools.core.install_context import RuntimeContext
+from lintro.tools.core.tool_installer import ToolInstaller
+from lintro.tools.core.tool_registry import ToolRegistry
+
+
+def _detect_project_languages() -> list[str]:
```

Ruff (0.15.7)

[warning] Too many branches (19 > 12)

(PLR0912)